### ISSUE TITLE
Issue with how IP Address is determined for Windows?

### ISSUE THREAD

**Issue by JCapriotti** [chef/ohai#746]
I'm experiencing an issue with how Ohai is returning the IP address of nodes in my Windows environment. Our nodes have multiple network interfaces, and it seems to be configured in [this manner](http://directaccess.richardhicks.com/2013/06/19/network-interface-configuration-for-multihomed-windows-server-2012-directaccess-servers/). In my case, using the article's lingo, we want Ohai to return the IP of the external adapter, and not the internal adapter.

I started poking at the source code and I'm not sure if it is functioning correctly, based on that article's configuration guidance. I've been focusing on this code section in [plugins/windows/network.rb](https://github.com/chef/ohai/blob/a15a010c351c12271d090a35c855c5fe8135297c/lib/ohai/plugins/windows/network.rb#L106-L109)

That code seems to be setting the default interface and gateway values to whatever the last adapter it loops over is, provided it contains a value for `default_ip_gateway`. However, when the default gateway for the "internal" adapter is configured according to that first link (not set), the value returned for `default_ip_gateway` is 0.0.0.0, which is not nil and has size > 0. 

It seems like the code is trying to only set the default interface when its gateway is set. I feel like it is incorrectly assuming the `default_ip_gateway` will return nil, but instead it is returning 0.0.0.0 and therefore incorrectly setting the default interface when the internal adapter happens to be the last in the loop.

I realize I'm a few layers deep in here... but hopefully I'm making some sense. I'm not great with configuring networking settings, but it does seem like the intent of the code agrees with that configuration article, it just isn't working...

Thoughts? I'd be happy to address this, but I'm just not clear on the intent of the code.

**Comment by lamont-granquist** [comment]
default gateway of '0.0.0.0' seems wrong.  i'd be concerned that this host got configured with a netmask of '0.0.0.0' instead of '255.255.255.0' or whatever and its ARP'ing on its subnet for the entire internet and has its own interface as its default gateway essentially.

of course we could just be buggy when we have two NICs as well...

**Comment by smurawski** [comment]
@JCapriotti I've tried setting up machine with multiple NICs and mulitple default gateways and I can't replicate the behavior.  

I also don't quite get why you would be getting 0.0.0.0 as a default gateway.  Could you share what version of ohai you are seeing this with, along with the output from ohai and `ipconfig /all` in a gist?

**Comment by JCapriotti** [comment]
@smurawski  Here you go: https://gist.github.com/JCapriotti/3792a0bf2b020aba7a31

I also included a [PowerShell snippet](https://gist.github.com/JCapriotti/3792a0bf2b020aba7a31#file-win32_networkadapterconfiguration-md) which is how I determined the last adapter in the list is returning 0.0.0.0 as the default gateway. My assumption here is the Ruby code will also return that value...

I mentioned it above, but based on how I am understanding [this code](https://github.com/chef/ohai/blob/a15a010c351c12271d090a35c855c5fe8135297c/lib/ohai/plugins/windows/network.rb#L106-L109), the if block will succeed twice. First for the adapter with IP 10.1.141.85, and second for the adapter with IP 10.1.141.101. The second hit will save that adapter's gateway as the default.
